File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.java
class Solution {
    public String solution(String code) {
        String answer = "";
        int mode = 0;
        for (int i = 0; i < code.length(); i++) {
            if (code.charAt(i) == '1')  {
                mode = 1 - mode;            //0->1 or 1->0
            } else if (i % 2 == mode) {     //mode==0 && idx%2==0 or mode==1 && idx%2==1
                answer += code.charAt(i);
            }
        }
        return answer.equals("") ? "EMPTY" : answer;
    }
}

File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.py
def solution(code):
    answer = ''
    mode = 0
    for i in range(len(code)):
        if code[i] == '1':
            mode = 1 - mode
        elif i % 2 == mode:
            answer += code[i]
    if not answer:
        answer = "EMPTY"
    return answer